cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -292,7 +292,12 @@ def _add_resources(self):
 
         # Budgets
         if self.config.dev_settings and self.config.dev_settings.budgets:
-            self.budgets = CostBudgets(self, self.config)
+            self.budgets = CostBudgets(
```

Comment:
NIT: since we have 4 parameters input here, it will be more readable if we add `cluster_config=self.config, BaseClusterConfig, is_batch=self._condition_is_batch(), create_lambda_role=self._condition_create_lambda_iam_role()` here

Reply:
Improved!